react/src/lib/components/Select/Select.tsx

```diff
@@ -149,31 +156,62 @@ export const Select: React.FC<InferProps<typeof propTypes>> = (
         parent_style,
         value,
         multi,
+        debounce_time_ms: debounce_time_ms,
         size,
         className,
         style,
         options,
         setProps,
     } = getPropsWithMissingValuesSetToDefault(props, defaultProps);
 
-    const handleChange = (e: React.ChangeEvent) => {
-        const selectedOptions = [].slice.call(
-            (e.target as HTMLSelectElement).selectedOptions
-        );
-        const values: (string | number)[] = [];
-
-        for (let i = 0; i < options.length; i++) {
-            if (
-                selectedOptions.some(
-                    (el: HTMLOptionElement) =>
-                        el.value === options[i].value.toString()
-                )
-            ) {
-                values.push(options[i].value);
-            }
+    const [selectedValues, setSelectedValues] =
+        React.useState<string | number | (string | number)[]>(value);
+
+    const debounceTimer =
+        React.useRef<ReturnType<typeof setTimeout> | null>(null);
+
+    React.useEffect(() => {
+        if (value !== selectedValues) {
+            setSelectedValues(value);
         }
-        setProps({ value: values });
-    };
+    }, [value]);
+
+    React.useEffect(() => {
+        // Unmount timer
+        return () => {
+            debounceTimer.current && clearTimeout(debounceTimer.current);
+        };
+    }, []);
+
+    const handleChange = React.useCallback(
+        (e: React.ChangeEvent) => {
+            const selectedOptions = [].slice.call(
+                (e.target as HTMLSelectElement).selectedOptions
+            );
+            const values: (string | number)[] = [];
+
+            for (let i = 0; i < options.length; i++) {
+                if (
+                    selectedOptions.some(
+                        (el: HTMLOptionElement) =>
+                            el.value === options[i].value.toString()
+                    )
+                ) {
+                    values.push(options[i].value);
+                }
+            }
+
+            if (values !== selectedValues) {
+                setSelectedValues(values);
+            }
```

Comment:
This will always return false since you're comparing two JavaScript object instances which are never the same. Compare:
```TypeScript
[] === [] // Will always return false
```
Consider using JSON.stringify(): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
or `lodash`: https://lodash.com/docs/#isEqual

Reply:
Using `lodash.isEqual`